Update Components with flow 3.0

ZheSun88 opened the pull request:
property `flow.version` should be updated in *flow-component-base*

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/vaadin-board-flow/110)
<!-- Reviewable:end -->

ZheSun88 committed:
Update Components with flow 3.0

web-padawan commented on vaadin-board-flow/src/main/java/com/vaadin/flow/component/board/internal/FunctionCaller.java:
```diff
@@ -40,7 +40,7 @@ public static void callOnceOnClientReponse(Component component,
         }
         element.setProperty(trackingProperty, true);
 
-        element.callFunction(function);
```
Let's split this PR to only include this line so that we could release a patch version.

IMO it would make sense to remove `HtmlImport` usage in a separate PR after that.

ZheSun88 replied:
Just include the api change doesn't fix the problem, as @HtmlImport has been removed from flow 3.0. 

the current validation build failed, because it is running against flow 2.X-SNAPSHOT, but not 3.0-snapshot. 
we need some more work to make a proper fix

tomivirkki replied:
Removing the @HtmlImport makes this a breaking change, so you'll also need to include some version bumps to pom.xml as well.

ZheSun88 replied:
for replacing the deprecated API, https://github.com/vaadin/vaadin-board-flow/pull/111

ZheSun88 replied:
> Removing the @HtmlImport makes this a breaking change, so you'll also need to include some version bumps to pom.xml as well.

this is the goal at the end.. I think for V15, we will need all the new major versions of components right?

tomivirkki replied:
Yes

DiegoCardoso committed:
Merge branch 'master' into update-with-flow-3.0

DiegoCardoso committed:
Upgrade flow-component-base and bump major

- Update interface signature
- Remove pom-bower-mode
- Remove webjar deps
- Update travis build
- Clean unused files on IT module

web-padawan reviewed (approved):
LGTM. 

It's weird that `vaadin-board-integration-tests/frontend/rowtypes.js` contained some CSS that was apparently there since SASS-based version, good that you now removed it.

DiegoCardoso commented:
It probably came from the migration from the V8 version and was kept unnoticed. The part I removed adds a white circle behind the numbers, but I'd rather remove it than try to fix + update all the screenshots. 😅

DiegoCardoso committed:
Remove IE11 and workaround horizontal scrollbar

DiegoCardoso committed:
Update .gitignore

DiegoCardoso committed:
Rollback profile name change

tomivirkki reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 13 of 15 files at r2, 7 of 7 files at r3.
> *[Reviewable](https://reviewable.io/reviews/vaadin/vaadin-board-flow/110#-:-LyiFpbl8MFCGVD06kKO:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->
